Plugins/GeoModel/include/Acts/Plugins/GeoModel/GeoModelDetectorObjectFactory.hpp

```diff
@@ -0,0 +1,78 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Detector/DetectorVolume.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Plugins/GeoModel/GeoModelDetectorElement.hpp"
+#include "Acts/Plugins/GeoModel/GeoModelToDetectorVolume.hpp"
+#include "Acts/Plugins/GeoModel/GeoModelTree.hpp"
+#include "Acts/Utilities/Logger.hpp"
+
+#include <GeoModelHelpers/getChildNodesWithTrf.h>
+
+#include "GeoModelKernel/GeoDefinitions.h"
+
+class GeoShape;
+struct GeoModelTree;
+class Surface;
+namespace Acts {
+class GeoModelDetectorObjectFactory {
+ public:
+  using GeoModelBoundingBox = std::shared_ptr<Experimental::DetectorVolume>;
+
+  struct Options {
+    std::vector<std::string> queries = {};
+  };
+  struct Config {
+    // /// List for names to match
+    std::vector<std::string> nameList;
+    /// List for materials to match
+    std::vector<std::string> materialList;
+
+    /// boolean flag to build subvolumes
+    bool convertSubVolumes = false;
+
+    /// flag to build the desired bounding Boxes
+    std::vector<std::string> convertBox;
+  };
+  struct Cache {
+    /// The created detector elements and their surfaces
+    std::vector<GeoModelSensitiveSurface> sensitiveSurfaces;
```

Comment:
I think the goal is to convert the FullPhysVols to Experimental::DetectorVolume containing the surfaces where ever bounding boxes are needed. Otherwise I considered the subvolumes being converted to SensitiveSurface as more of an artifact of the converter then anything else. So as far as I can tell there is no need to use SensitiveSurface, I just did it because this is the return type of the surface converter